test/tools/yulopti.cpp

```diff
@@ -162,90 +172,114 @@ class YulOpti
 		}
 	}
 
-	void runInteractive(string source)
+	void disambiguate()
 	{
-		bool disambiguated = false;
+		*m_ast = std::get<yul::Block>(Disambiguator(m_dialect, *m_analysisInfo)(*m_ast));
+		m_analysisInfo.reset();
+		m_nameDispenser.reset(*m_ast);
+	}
+
+	bool runSteps(string _source, string _steps)
+	{
+		if (!parse(_source))
+			return false;
+		try
+		{
+			disambiguate();
+			OptimiserSuite{m_context}.runSequence(_steps, *m_ast);
+			cout << AsmPrinter{m_dialect}(*m_ast) << endl;
+		}
+		catch (...)
+		{
+			cerr << endl << "Exception:" << endl;
+			cerr << boost::current_exception_diagnostic_information() << endl;
+			return false;
+		}
```

Comment:
I'd put this handler over the whole `main()` instead.

Other than the loop (which must not exit on errors), I think that nothing in this file should have a `(...)` handler. There should be only a single top-level handler that catches anything not caught lower, prints a message and exits.

Reply:
We also won't get rid of catching in ``parse`` at least, because that needs the error reporter, if it's supposed to report anything meaningful...

Reply:
True. I mean, I think `parse()` should catch it, print the errors from error reporter and then rethrow it to the top-level handler but that won't let us get rid of the extra handler unfortunately.